full_bubble_model.py

```diff
@@ -15,20 +19,104 @@
 try:
     import parameters as par   # numeric constants and coefficents
     importlib.reload(par)   # reload changes you made
-    print(f'model: {par.model}')
 except:
-    print('Error, \'parameters.py\' not found')
+    print(print(colored('Error, \'parameters.py\' not found','red')))
+try:
+    import excitation
+    importlib.reload(excitation)
+except:
+    try:
+        import Bubble_dynamics_simulation.excitation as excitation
+        importlib.reload(excitation)
+    except:
+        print(colored(f'Error, \'excitation.py\' not found', 'red'))    
+
+"""________________________________Settings________________________________"""
+
+enable_heat_transfer = True
+enable_evaporation = False
+enable_reactions = True
+enable_dissipated_energy = False
+target_specie = 'NH3' # Specie to calculate energy effiqiency
+excitation_type = 'no_excitation' # function to calculate pressure excitation
+
+"""________________________________General________________________________"""
 
-# dot.notation access to dictionary attributes
-# instead of dictionary['key'] you can use dictionary.key
 class dotdict(dict):
+    """Dot notation access to dictionary attributes. 
+    Instead of dictionary['key'] you can use dictionary.key"""
     __getattr__ = dict.get
     __setattr__ = dict.__setitem__
     __delattr__ = dict.__delitem__
-    
+
+def copy(input):
+    """deep copy of input"""
+
+    if type(input) == list:
+        return [copy(element) for element in input]
+    elif type(input) == dict:
+        return {key: copy(value) for key, value in input.items()}
+    elif type(input) == np.ndarray:
+        return input.copy()
+    elif type(input) == dotdict:
+        return dotdict({key: copy(value) for key, value in input.items()})
+    else:
+        return input
 
 """________________________________Before the simulation________________________________"""
 
+def colorTF(boolean):
+    return colored(str(boolean), 'green') if boolean else colored(str(boolean), 'red')
+
+Excitation, excitation_args, excitation_units = excitation.getExcitation(excitation_type=excitation_type)
+if par.indexOfWater == -1:
+    enable_evaporation = False
+print(f'model: {par.model}')
+print(f'target specie: {target_specie}')
+print(f'excitation: {excitation_type} (control parameters: {excitation_args})')
+print(f'enable heat transfer: {colorTF(enable_heat_transfer)}\tenable evaporation: {colorTF(enable_evaporation)}\tenable reactions: {colorTF(enable_reactions)}\tenable dissipated energy: {colorTF(enable_dissipated_energy)}')
+if target_specie not in par.species:
+    print(colored(f'Error, target specie \'{target_specie}\' not found in parameters.py', 'red'))
+    
+def example_cpar(normal_dict=False):
+    '''Provides an example of the control parameter dictionary. Use print_cpar() to print it. Parameters:
```

Comment:
Provides an example of the control parameter dictionary with default values. Can be useful and speed up work. Use print_cpar() to display it in an organized manner. The result is:
cpar = de.dotdict(dict(
    ID = 0,                                      # ID of control parameter (not used during calculation)
  # Initial conditions:
    R_E =  0.00001000,                           # bubble equilibrium radius [m]
    ratio =  1.00,                               # initial radius / equilibrium radius R_0/R_E [-]
    gases = [par.index['H2'], par.index['N2']],  # indexes of species in initial bubble (list of species indexes)
    fractions = [0.75, 0.25],                    # molar fractions of species in initial bubble (list of fractions for every gas)
  # Ambient parameters:
    P_amb =  101325.00,                          # ambient pressure [Pa]
    T_inf =  293.15,                             # ambient temperature [K]
  # Liquid parameters:
    alfa_M =  0.3500,                            # water accommodation coefficient [-]
    P_v =  2338.10,                              # vapour pressure [Pa]
    mu_L =  0.0010,                              # dynamic viscosity [Pa*s]
    c_L =  1483.00,                              # sound speed [m/s]
    surfactant =  1.00,                          # surfactant (surface tension modfier) [-]
  # Excitation parameters: (excitation_type = no_excitation)
))

# Calculate pressure/temperature dependent parameters:
cpar.mu_L = de.Viscosity(cpar.T_inf)
cpar.P_v = de.VapourPressure(cpar.T_inf)